// stripe/lib/resources.js
'use strict';

const resourceNamespace = require('stripe/lib/ResourceNamespace');

module.exports = {
  Accounts: require('stripe/lib/resources/Accounts'),
  // Support Accounts for consistency, Account for backwards compatibility
  Account: require('stripe/lib/resources/Accounts'),
  AccountLinks: require('stripe/lib/resources/AccountLinks'),
  ApplePayDomains: require('stripe/lib/resources/ApplePayDomains'),
  ApplicationFees: require('stripe/lib/resources/ApplicationFees'),
  Balance: require('stripe/lib/resources/Balance'),
  BalanceTransactions: require('stripe/lib/resources/BalanceTransactions'),
  BitcoinReceivers: require('stripe/lib/resources/BitcoinReceivers'),
  Charges: require('stripe/lib/resources/Charges'),
  CountrySpecs: require('stripe/lib/resources/CountrySpecs'),
  Coupons: require('stripe/lib/resources/Coupons'),
  CreditNotes: require('stripe/lib/resources/CreditNotes'),
  Customers: require('stripe/lib/resources/Customers'),
  Disputes: require('stripe/lib/resources/Disputes'),
  EphemeralKeys: require('stripe/lib/resources/EphemeralKeys'),
  Events: require('stripe/lib/resources/Events'),
  ExchangeRates: require('stripe/lib/resources/ExchangeRates'),
  Files: require('stripe/lib/resources/Files'),
  FileLinks: require('stripe/lib/resources/FileLinks'),
  Invoices: require('stripe/lib/resources/Invoices'),
  InvoiceItems: require('stripe/lib/resources/InvoiceItems'),
  IssuerFraudRecords: require('stripe/lib/resources/IssuerFraudRecords'),
  Mandates: require('stripe/lib/resources/Mandates'),
  OAuth: require('stripe/lib/resources/OAuth'),
  Orders: require('stripe/lib/resources/Orders'),
  OrderReturns: require('stripe/lib/resources/OrderReturns'),
  PaymentIntents: require('stripe/lib/resources/PaymentIntents'),
  PaymentMethods: require('stripe/lib/resources/PaymentMethods'),
  Payouts: require('stripe/lib/resources/Payouts'),
  Plans: require('stripe/lib/resources/Plans'),
  Products: require('stripe/lib/resources/Products'),
  Recipients: require('stripe/lib/resources/Recipients'),
  Refunds: require('stripe/lib/resources/Refunds'),
  Reviews: require('stripe/lib/resources/Reviews'),
  SetupIntents: require('stripe/lib/resources/SetupIntents'),
  Skus: require('stripe/lib/resources/SKUs'),
  Sources: require('stripe/lib/resources/Sources'),
  Subscriptions: require('stripe/lib/resources/Subscriptions'),
  SubscriptionItems: require('stripe/lib/resources/SubscriptionItems'),
  SubscriptionSchedules: require('stripe/lib/resources/SubscriptionSchedules'),
  TaxRates: require('stripe/lib/resources/TaxRates'),
  ThreeDSecure: require('stripe/lib/resources/ThreeDSecure'),
  Tokens: require('stripe/lib/resources/Tokens'),
  Topups: require('stripe/lib/resources/Topups'),
  Transfers: require('stripe/lib/resources/Transfers'),
  UsageRecords: require('stripe/lib/resources/UsageRecords'),
  UsageRecordSummaries: require('stripe/lib/resources/UsageRecordSummaries'),
  WebhookEndpoints: require('stripe/lib/resources/WebhookEndpoints'),
  Checkout: resourceNamespace('checkout', {
    Sessions: require('stripe/lib/resources/Checkout/Sessions'),
  }),
  Issuing: resourceNamespace('issuing', {
    Authorizations: require('stripe/lib/resources/Issuing/Authorizations'),
    Cards: require('stripe/lib/resources/Issuing/Cards'),
    Cardholders: require('stripe/lib/resources/Issuing/Cardholders'),
    Disputes: require('stripe/lib/resources/Issuing/Disputes'),
    Transactions: require('stripe/lib/resources/Issuing/Transactions'),
  }),
  Radar: resourceNamespace('radar', {
    EarlyFraudWarnings: require('stripe/lib/resources/Radar/EarlyFraudWarnings'),
    ValueLists: require('stripe/lib/resources/Radar/ValueLists'),
    ValueListItems: require('stripe/lib/resources/Radar/ValueListItems'),
  }),
  Reporting: resourceNamespace('reporting', {
    ReportRuns: require('stripe/lib/resources/Reporting/ReportRuns'),
    ReportTypes: require('stripe/lib/resources/Reporting/ReportTypes'),
  }),
  Sigma: resourceNamespace('sigma', {
    ScheduledQueryRuns: require('stripe/lib/resources/Sigma/ScheduledQueryRuns'),
  }),
  Terminal: resourceNamespace('terminal', {
    ConnectionTokens: require('stripe/lib/resources/Terminal/ConnectionTokens'),
    Locations: require('stripe/lib/resources/Terminal/Locations'),
    Readers: require('stripe/lib/resources/Terminal/Readers'),
  }),
};
